pandas/core/indexes/numeric.py

```diff
@@ -367,12 +367,13 @@ def astype(self, dtype, copy=True):
                 "values are required for conversion"
             ).format(dtype=dtype)
             raise TypeError(msg)
-        elif (
-            is_integer_dtype(dtype) and not is_extension_array_dtype(dtype)
-        ) and self.hasnans:
+        elif is_integer_dtype(dtype) and not is_extension_array_dtype(dtype):
             # TODO(jreback); this can change once we have an EA Index type
             # GH 13149
-            raise ValueError("Cannot convert NA to integer")
+            if self.hasnans:
```

Comment:
Can this re-use `astype_nansafe`? That should handle things, including the nullable-integer check.